presto-main/src/main/java/com/facebook/presto/execution/SqlQueryExecution.java

```diff
@@ -437,12 +456,15 @@ public void fail(Throwable cause)
     }
 
     @Override
-    public Duration waitForStateChange(QueryState currentState, Duration maxWait)
-            throws InterruptedException
+    public ListenableFuture<QueryOutputInfo> getOutputInfo()
     {
-        try (SetThreadName ignored = new SetThreadName("Query-%s", stateMachine.getQueryId())) {
-            return stateMachine.waitForStateChange(currentState, maxWait);
-        }
+        return Futures.nonCancellationPropagating(outputInfo);
```

Comment:
The point is I get to choose when to disable cancellation...  Futures are always cancelable and in this case I want to prevent the caller from canceling my future.